utils/fs-cache.js

```diff
@@ -0,0 +1,111 @@
+// @ts-check
+const { join } = require("path");
+const { mkdir, unlink } = require("fs").promises;
+const { createHash } = require("crypto");
+const { readJSON, writeJSON, readDir } = require("./fs.js");
+const { MemCache } = require("./mem-cache.js");
+
+/**
+ * An in-memory cache persisted by file-system.
+ * @template ValueType
+ */
+class FsCache {
+  /** @type {string} */
+  #dirname = null;
+  /** @type {MemCache<ValueType>} */
+  #cache = null;
+  /** @type {string[]} */
+  #saveQueue = [];
+  /** @type {string[]} */
+  #deleteQueue = [];
+
+  /**
+   * @param {string} dirname where to store cached items on file-system
+   * @typedef {{ key: string, time: number, value: ValueType }} FsCacheEntry
+   */
+  constructor(dirname, ttl = Infinity, autosaveInterval = 10000) {
+    this.#dirname = dirname;
+    this.#cache = new MemCache(ttl);
+    if (autosaveInterval > 0) {
+      setTimeout(this.dump.bind(this), autosaveInterval);
```

Comment:
Just wondering, what's the rationale of using the deleted file count here?  seems a bit arbitrary is all.

Reply:
Um.. this comment seems to be out of context.

Reply:
oops, sorry... I read the function backwards... the correct question is, why does .dump() return a number?